multitenancy/src/main/java/org/axonframework/extensions/multitenancy/components/deadletterqueue/MultiTenantDeadLetterQueue.java

```diff
@@ -0,0 +1,406 @@
+        if (currentTenant != null) {
+            return fetchFromTenantSegment(currentTenant, SequencedDeadLetterQueue::amountOfSequences);
+        } else {
+            logger.info("No tenant found for current thread. Returning total amount of sequences.");
+            return tenants.stream().mapToLong(tenant -> fetchFromTenantSegment(tenant,
+                                                                               SequencedDeadLetterQueue::amountOfSequences))
+                          .sum();
+        }
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public boolean process(Predicate<DeadLetter<? extends M>> sequenceFilter,
+                           Function<DeadLetter<? extends M>, EnqueueDecision<M>> processingTask) {
+        TenantDescriptor currentTenant = TenantWrappedTransactionManager.getCurrentTenant();
+        if (currentTenant != null) {
+            return fetchFromTenantSegment(currentTenant, seg -> seg.process(sequenceFilter, processingTask));
+        } else {
+            logger.info("No tenant found for current thread. Processing all tenants queues.");
+            return tenants.stream().map(tenant -> fetchFromTenantSegment(tenant,
+                                                                         seg -> seg.process(sequenceFilter,
+                                                                                            processingTask))).reduce(
+                    false,
+                    (a, b) -> a || b);
+        }
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public boolean process(Function<DeadLetter<? extends M>, EnqueueDecision<M>> processingTask) {
+        TenantDescriptor currentTenant = TenantWrappedTransactionManager.getCurrentTenant();
+        if (currentTenant != null) {
+            return fetchFromTenantSegment(currentTenant, seg -> seg.process(processingTask));
+        } else {
+            logger.info("No tenant found for current thread. Processing all tenants queues.");
+            return tenants.stream().map(tenant -> fetchFromTenantSegment(tenant, seg -> seg.process(processingTask)))
+                          .reduce(false, (a, b) -> a || b);
+        }
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public void clear() {
+        TenantDescriptor currentTenant = TenantWrappedTransactionManager.getCurrentTenant();
+        if (currentTenant != null) {
+            executeForTenantSegment(currentTenant, SequencedDeadLetterQueue::clear);
+        } else {
+            logger.info("No tenant found for current thread. Clearing all tenants queues.");
+            tenants.forEach(tenant -> executeForTenantSegment(tenant, SequencedDeadLetterQueue::clear));
+        }
+    }
+
+    private <R> R fetchFromTenantSegment(TenantDescriptor tenantDescriptor,
+                                         Function<SequencedDeadLetterQueue<M>, R> function) {
```

Comment:
Would you mind renaming the `function` parameter to something a bit more descriptive?